### ISSUE TITLE
生成背景是绿色

### ISSUE THREAD

**Issue by pangyunping** [kaonashi-tyc/zi2zi#38]
你好，我训练了之后打印出来的生成图背景是绿色的，可以请教一下原因吗？
未曾改过代码，邮箱联系一下吧，谢谢
1054778289@qq.com

**Comment by pangyunping** [comment]
训练以后效果还是没有作者这么好

**Comment by fanxuedong6** [comment]
兄弟 你训练了多久 我这16g+1080ti+i7 训练了30个小时了还没出结果

**Comment by Nicersmeng** [comment]
我生成出来时紫色，你问题解决了吗？

**Comment by miraclezqc** [comment]
你好 我想问一下你们可以拿作者的font27的模型继续训练吗 总是报错  需要一些特别的设置吗？

**Comment by H-xie** [comment]
当载入checkpoint失败时，infer得到的图像底色会很奇怪（我不认识那个颜色）
注意！infer未能载入checkpoint的时候不会中断，只会输出一行提示
```fail to restore model XXXX/checkpoint```

在训练时，程序保存checkpoint的路径会更深，打开`XXXX/checkpoint` 文件夹，infer前需要看一下。
当正确载入模型时，提示
```restored model XXXX/checkpoint/experiment_0_batch_16```
可以得到预期的图像**白底黑字**

**Comment by ouguozhen** [comment]
@H-xie 兄弟,你是怎么构建训练集的呢?只是运行一遍font2img.py 和pacage.py是不行的吧?假如说我想训练关于A,B,C,D四种字体的模型,那么我应该怎么去构建训练集呢?

**Comment by cinastanbean** [comment]
Hey, I got this fucking bug too. You can fix it as @H-xie 

"restored model ./experiment/checkpoint/experiment_0_batch_16"

**Comment by chenjiechenjiejiejie** [comment]
> 我生成出来时紫色，你问题解决了吗？


> 当载入关卡失败时，推断得到的图像底色会很奇怪（我不认识那个颜色）
> 注意！推断未能载入关卡的时候不会中断，只会输出一行提示
> `fail to restore model XXXX/checkpoint`
> 
> 在训练时，程序保存检查点的路径会更深，打开`XXXX/checkpoint`文件夹数，推断前需要看一下。
> 当正确载入模型时，提示
> `restored model XXXX/checkpoint/experiment_0_batch_16`
> 可以得到预期的图像**白底黑字**

这个具体怎么解决呢？？还是没有解决了。

**Comment by not-bald-owl** [comment]
这是我推理中出现的运行结果：

restored model experiment/checkpoint/experiment_0_batch_16/
2024-03-29 13:20:35.823419: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape ingenerator_1/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer
2024-03-29 13:20:36.757899: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8700
2024-03-29 13:20:37.257713: I tensorflow/tsl/platform/default/subprocess.cc:304] Start cannot spawn child process: No such file or directory
2024-03-29 13:20:38.277277: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
Image saved successfully to inference_results2/inferred_0009.png
generated images saved at inference_results2/inferred_0009.png
Image saved successfully to inference_results2/inferred_0019.png
generated images saved at inference_results2/inferred_0019.png
Image saved successfully to inference_results2/inferred_0029.png
generated images saved at inference_results2/inferred_0029.png
Image saved successfully to inference_results2/inferred_0039.png
generated images saved at inference_results2/inferred_0039.png
Image saved successfully to inference_results2/inferred_0049.png
generated images saved at inference_results2/inferred_0049.png
Image saved successfully to inference_results2/inferred_0052.png
generated images saved at inference_results2/inferred_0052.png

下面粘上我的推理代码（楼上说的对 要改个模型路径）：
`python infer.py --model_dir=experiment/checkpoint/experiment_0_batch_16/  \
                --batch_size=16 \
                --source_obj=experiment/data/train.obj \
                --embedding_ids=0 \
                --save_dir=inference_results2/`
                
 但是我训练之后得到的背景图是紫色的 很奇怪 请问有大佬解答吗
 
![image](https://github.com/kaonashi-tyc/zi2zi/assets/120432985/9e475ecd-a1a1-432f-9c5a-e3116ce24687)

**Comment by not-bald-owl** [comment]
> 我生成出来时紫色，你问题解决了吗？

同是紫色 你解决了吗
